packages/client/src/pages/RecommendationsPage/RecommendationsPage.tsx

```diff
@@ -73,7 +73,14 @@ const RecommendationsPage = (): ReactElement => {
     params: GridRowParams,
     _event: MuiEvent<SyntheticEvent>,
   ) => {
-    setSelectedRecommendation(params.row as RecommendationRow)
+    if (
+      selectedRecommendation &&
+      params.row.accountId === selectedRecommendation.accountId
```

Comment:
Sometimes there can be multiple recommendations listed that happen to apply to the same account. This could cause the panel to unwillingly close instead of updating when clicking on a new row that has a different recommendation for that same account. This is due to the accountId's being the same, despite the recommendation information being different. The Material UI DataGrid assigns unique ID's to its rows that resembled a unique index. I would recommend changing this line to check for that instead 😄 
```suggestion
      params.row.id === selectedRecommendation.id
```

Reply:
Yes you're spot on, great catch! Have applied your suggestion and the tests still pass. Have left the tests the way they are as it's a bit awkward to generate a specific test case for this and maybe not worth the hassle. 
Some sort of faker-like library to generate random test cases with parameters to overwrite would probably be useful here.